core/test/org/javarosa/core/util/test/LocalizationTests.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2009 JavaRosa
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.javarosa.core.util.test;
+
+import org.javarosa.core.services.locale.LocalizationUtils;
+import org.javarosa.core.util.OrderedHashtable;
+
+import j2meunit.framework.Test;
+import j2meunit.framework.TestCase;
+import j2meunit.framework.TestMethod;
+import j2meunit.framework.TestSuite;
+
+public class LocalizationTests extends TestCase {
+
+    private static int NUM_TESTS = 2;
+
+    /* (non-Javadoc)
+     * @see j2meunit.framework.TestCase#setUp()
+     */
+    protected void setUp() throws Exception {
+        super.setUp();
+    }
+
+    public LocalizationTests(String name, TestMethod rTestMethod) {
+        super(name, rTestMethod);
+    }
+
+    public LocalizationTests(String name) {
+        super(name);
+    }
+
+    public LocalizationTests() {
+        super();
+    }
+
+    public Test suite() {
+        TestSuite aSuite = new TestSuite();
+
+        for (int i = 1; i <= NUM_TESTS; i++) {
+            final int testID = i;
+
+            aSuite.addTest(new LocalizationTests("Localization Test " + i, new TestMethod() {
+                public void run(TestCase tc) {
+                    ((LocalizationTests)tc).testMaster(testID);
+                }
+            }));
+        }
+
+        return aSuite;
+    }
+
+    public void testMaster(int testID) {
+        //System.out.println("running " + testID);
+
+        switch (testID) {
+            case 1:
+                testParseValue();
+                break;
+            case 2:
+                testParseAndAdd();
+                break;
+        }
+    }
+
+    public void testParseValue() {
+
+        String result1 = "";
+        String result2 = "";
+
+        result1 = LocalizationUtils.parseValue("1. One \\n2. Two \\n3. Three");
+        result2 = LocalizationUtils.parseValue("\\# Start");
+
+        assertTrue("Parsed newlines correctly", result1.equals("1. One \n2. Two \n3. Three"));
+        assertTrue("Parsed newlines correctly", result2.equals("# Start"));
+    }
+
+    public void testParseAndAdd(){
+
+        OrderedHashtable<String, String> testTable = new OrderedHashtable<String, String>();
+
+        //LocalizationUtils.parseAndAdd(testTable,"string.0=#this line should be all comment", 0);
```

Comment:
Why did this line get removed? It seems like the we should have a test that starts with "#" just to make sure it doesn't crash anything.

like

"#comment goes here"